Lib/_strptime.py

```diff
@@ -586,3 +591,32 @@ def _strptime_datetime(cls, data_string, format="%a %b %d %H:%M:%S %Y"):
         args += (tz,)
 
     return cls(*args)
+
+def _strptime_datetime_date(data_string, format):
+    """Return a date based on the input string and the format string."""
+    if not format:
+        raise ValueError("Date format is not valid.")
+    msg = "'{!s}' {} not valid in date format specification."
+    if _check_invalid_datetime_specs(format, time_specs, msg):
+        _date = _strptime_datetime(datetime_datetime, data_string, format)
+        return _date.date()
+
+def _strptime_datetime_time(data_string, format):
+    """Return a time based on the input string and the format string."""
+    if not format:
+        raise ValueError("Date format is not valid.")
+    msg = "'{!s}' {} not valid in time format specification."
+    if _check_invalid_datetime_specs(format, date_specs, msg):
+        _time = _strptime_datetime(datetime_datetime, data_string, format)
+        return _time.time()
+
+def _check_invalid_datetime_specs(fmt, specs, msg):
```

Comment:
If I'm understanding this correctly, this function seems to have the inverted sense of what I would expect. It seems that it checks if `fmt` is a **valid** spec.

From the names of the variables and functions, I was thinking that this would be a whitelist not a blacklist. Does it make sense to switch to a whitelist approach? If not, can we maybe change `specs` to be `blacklist_specs` or something?